url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/robot/SOCRobotBrain.java

```diff
@@ -0,0 +1,5460 @@
+     * @return our player data
+     */
+    public SOCPlayer getOurPlayerData()
+    {
+        return ourPlayerData;
+    }
+
+    /**
+     * @return the building plan
+     */
+    public Stack getBuildingPlan()
+    {
+        return buildingPlan;
+    }
+
+    /**
+     * @return the decision maker
+     */
+    public SOCRobotDM getDecisionMaker()
+    {
+        return decisionMaker;
+    }
+
+    /**
+     * turns the debug recorders on
+     */
+    public void turnOnDRecorder()
+    {
+        dRecorder[0].turnOn();
+        dRecorder[1].turnOn();
+    }
+
+    /**
+     * turns the debug recorders off
+     */
+    public void turnOffDRecorder()
+    {
+        dRecorder[0].turnOff();
+        dRecorder[1].turnOff();
+    }
+
+    /**
+     * @return the debug recorder
+     */
+    public DebugRecorder getDRecorder()
+    {
+        return dRecorder[currentDRecorder];
+    }
+
+    /**
+     * @return the old debug recorder
+     */
+    public DebugRecorder getOldDRecorder()
+    {
+        return dRecorder[(currentDRecorder + 1) % 2];
+    }
+
+    /**
+     * @return the last move we made
+     */
+    public SOCPossiblePiece getLastMove()
+    {
+        return lastMove;
+    }
+
+    /**
+     * @return our last target piece
+     */
+    public SOCPossiblePiece getLastTarget()
+    {
+        return lastTarget;
+    }
+
+    /**
+     * Find our player data using our nickname
+     */
+    public void setOurPlayerData()
+    {
+        ourPlayerData = game.getPlayer(client.getNickname());
+        ourPlayerTracker = new SOCPlayerTracker(ourPlayerData, this);
+        int opn = ourPlayerData.getPlayerNumber();
+        playerTrackers = new HashMap();
+        playerTrackers.put(new Integer(opn), ourPlayerTracker);
+
+        for (int pn = 0; pn < game.maxPlayers; pn++)
+        {
+            if ((pn != opn) && ! game.isSeatVacant(pn))
+            {
+                SOCPlayerTracker tracker = new SOCPlayerTracker(game.getPlayer(pn), this);
+                playerTrackers.put(new Integer(pn), tracker);
+            }
+        }
+
+        decisionMaker = new SOCRobotDM(this);
+        negotiator = new SOCRobotNegotiator(this);
+        dummyCancelPlayerData = new SOCPlayer(-2, game);
+
+        // Verify expected face (fast or smart robot)
+        int faceId;
+        switch (getRobotParameters().getStrategyType())
+        {
+        case SOCRobotDM.SMART_STRATEGY:
+            faceId = -1;  // smarter robot face
+            break;
+
+        default:
+            faceId = 0;   // default robot face
+        }
+        if (ourPlayerData.getFaceId() != faceId)
+        {
+            ourPlayerData.setFaceId(faceId);
+            // robotclient will handle sending it to server
+        }
+    }
+
+    /**
+     * Here is the run method.  Just keep receiving game events
+     * and deal with each one.
+     * Remember that we're sent a {@link SOCGameTextMsg}(<tt>"*PING*"</tt>) once per second.
+     */
+    public void run()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 112. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.